iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstConverter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.helm.tree.utils;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.AnyOrBuilder;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.MessageOrBuilder;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.CheckForNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.iac.helm.tree.Node;
+
+public final class GoTemplateAstConverter {
+  private static final Logger LOG = LoggerFactory.getLogger(GoTemplateAstConverter.class);
+  private static final MethodHandles.Lookup lookup = MethodHandles.lookup();
+
+  private GoTemplateAstConverter() {
+    // utility class
+  }
+
+  @CheckForNull
+  public static Node unpackNode(Any nodePb) {
+    try {
+      var types = typesForMessage(nodePb);
+      Class<? extends GeneratedMessageV3> messageClass = types.messageType;
+      Class<? extends MessageOrBuilder> messageClassOrBuilder = types.messageOrBuilderType;
+      Class<? extends Node> targetClass = types.nodeType;
+      var handle = lookup.findStatic(targetClass, "fromPb", MethodType.methodType(Node.class, messageClassOrBuilder));
+      return (Node) handle.invoke(nodePb.unpack(messageClass));
+    } catch (Throwable t) {
+      LOG.debug("Failed to unpack node", t);
+      return null;
+    }
+  }
+
+  public static List<Node> unpack(Collection<Any> nodesPb) {
+    return nodesPb.stream()
+      .map(GoTemplateAstConverter::unpackNode)
+      .collect(Collectors.toList());
+  }
+
+  private static String typeName(AnyOrBuilder nodePb) {
+    return nodePb.getTypeUrl().substring(nodePb.getTypeUrl().lastIndexOf('/') + 1);
+  }
+
+  private static Types typesForMessage(AnyOrBuilder nodePb) {
+    Types types;
+    switch (typeName(nodePb)) {
+      case "org.sonar.iac.helm.ActionNode":
+        types = new Types(org.sonar.iac.helm.ActionNode.class, org.sonar.iac.helm.ActionNodeOrBuilder.class, org.sonar.iac.helm.tree.ActionNode.class);
```

Comment:
Consider also moving generated protobuf code to separate package `*.proto.*` or `*.protobuf.*` for better readability and separation of code.

Reply:
Done in c8f2746